codegen/src/types/type_path.rs

```diff
@@ -76,6 +76,27 @@ impl TypePath {
             Self::Substitute(sub) => sub.parent_type_params(acc),
         }
     }
+
+    /// Gets the vector type parameter if the data is represented as `TypeDef::Sequence`.
+    ///
+    /// **Note:** Utilized for transforming `std::vec::Vec<T>` into slices `&[T]` for the storage API.
+    pub fn vec_type_param(&self) -> Option<TokenStream> {
+        match self {
+            TypePath::Type(ty) => {
+                match ty.ty.type_def() {
+                    TypeDef::Sequence(_) => {
+                        if ty.params.is_empty() {
```

Comment:
Looking at the `scale-info` crate it looks like you could more safely do:

```
TypeDef::Sequence(seq) => {
    let type_param = seq.type_param()
    // ...
```

I'd also lean towards not bothering to `quote!` the type_param before returning it (I think it'll work just fine if you don't), but I'm not too bothered either way!

Also as a matter of personal taste I'd probably avoid the indentation with some early returns, a bit like:

```
let ty = match self {
    TypePath::Type(ty) => ty,
    _ => return None
};

let sequence = match ty.ty.type_def() {
    TypeDef::Sequence(seq) => seq,
    _ => return None;
};

// ...
```

Reply:
That makes a lot of sense, and indeed the code looks a lot better 😄 

I did give the `type_param()` a try, but it seems to return a `UntrackedSymbol<TypeId> ` which does not implement the `ToToken` trait.
Looking below in the code generation, it seems under `TypeDef::Sequence` there is the assumption of always having at least one type_param. I've removed the `is_empty()` check here for consistency.
Wondering if we could run into problems along the line due to this assumption. 🤔